test/common/router/config_impl_test.cc

```diff
@@ -3975,6 +3977,20 @@ name: AllRedirects
     redirect->rewritePathHeader(headers);
     EXPECT_EQ("https://redirect.lyft.com/https/prefix/", redirect->newPath(headers));
   }
+  {
+    Http::TestHeaderMapImpl headers =
+        genRedirectHeaders("redirect.lyft.com", "/ignore-this", false, false);
+    const DirectResponseEntry* redirect = config.route(headers, 0)->directResponseEntry();
+    redirect->rewritePathHeader(headers);
+    EXPECT_EQ("http://redirect.lyft.com/", redirect->newPath(headers));
+  }
+  {
```

Comment:
It's worth some comments here to describe what how these two cases are different.

Reply:
OK, will add some comments here.